.github/workflows/assets-test.yaml

```diff
@@ -138,7 +138,19 @@ jobs:
           tenant-id: ${{ secrets.AZURE_TENANT_ID }}
           subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
           scripts-setup-dir: ${{ env.scripts_setup_dir }}
-      
+
+      - name: Fetch OID token every 4 mins
+        run: |
+          while true; do
+            token_request=$ACTIONS_ID_TOKEN_REQUEST_TOKEN
+            token_uri=$ACTIONS_ID_TOKEN_REQUEST_URL
+            token=$(curl -H "Authorization: bearer $token_request" "${token_uri}&audience=api://AzureADTokenExchange" | jq .value -r)
```

Comment:
It would help to log the expiration of the federated token here, and ideally use the value to control the sleep timer. Looks like [the federated token is a JWT](https://github.com/Azure/login/blob/e9468bad0a1e2a5863b25f77ea9b6ce0d19f5e41/src/common/LoginConfig.ts#L85).

Reply:
Unfortunately, I couldn't see the expiry key in the output from the curl command. It just had "value" and "count" keys in the json.

Reply:
You'd have to parse the JWT in the value to see it.

Reply:
Thanks @elesel

![image](https://github.com/Azure/azureml-assets/assets/74648335/45f665f0-5fe2-46d8-adf9-e1bc87a3e2a4)